### ISSUE TITLE
sorting by custom attributes

### ISSUE THREAD

**Issue by senid231** [JSONAPI-Resources/jsonapi-resources#936]
For example we have such code
```ruby
class Person < ApplicationRecord
  has_one :parent_person, class_name: 'Person'
  has_one :detail, class_name: 'PersonDetail'

  alias_attribute :active, :is_active

  def independent
    self.parent_person_id.nil?
  end

  def ensure_detail
    self.detail || self.build_detail
  end
end

class PersonDetail < ApplicationRecord
  belongs_to :person
end

class PersonResource < ApplicationResource
  attributes :email, :address, :is_active, :independent
  filter :address, apply: ->(value) { where(person_details: {address: value}) }
  filter :independent, apply: ->(value) do
    flag = !(value == 'false')
    where(flag ? 'parent_person_id IS NULL' : 'parent_person_id IS NOT NULL')
  end

  def address
    @model.detail.address
  end

  def address=(value)
    @model.ensure_detail.address = value
  end

  def self.creatable_fields(context = nil)
    super - [:active, :independent]
  end

  def self.updatable_fields(context = nil)
    super - [:active, :independent]
  end

  def self.records(options = {})
    super.eager_load(:detail)
  end
end
```

and we want to sort by `address`, `is_active` and `independent`

in past version of `JR` I use code like this:
```ruby
def custom_sorting
  {
    'address' => 'person_details.address',
    'is_active' => 'active',
    'independent' => '(parent_person_id IS NULL)'
  }
end

def self.apply_sort(records, order_options, context = {})
  changed_order_options = []
  order_options.each do |key, value|
    changed_order_options << "#{custom_sorting.fetch(key,key)} #{value}"
  end
  super(records, changed_order_options, context)
end
```

but after PR #494 it doesn't work

Do you have any idea how can we now sort by custom attributes?

**Comment by adambedford** [comment]
@senid231 Your custom sorting work looks awesome. I'd love the ability to define `sort`s in the same way as `filter`s. How do we get this into core?

**Comment by senid231** [comment]
@adambedford there are many bugs in current `master` so we need to fix them at first

I think this PR will be merged before next stable release comes

**Comment by adambedford** [comment]
@lgebhardt @senid231 Any clue on when this will be included in a release?

**Comment by senid231** [comment]
@adambedford was merged to master in PR #1109
